cpp/ql/src/Critical/OverflowStatic.ql

```diff
@@ -94,10 +97,15 @@ class CallWithBufferSize extends FunctionCall {
   }
 
   int statedSizeValue() {
-    exists(Expr statedSizeSrc |
-      DataFlow::localExprFlow(statedSizeSrc, statedSizeExpr()) and
-      result = statedSizeSrc.getValue().toInt()
-    )
+    // `upperBound(e)` defaults to `exprMaxVal(e)` when `e` isn't analyzable. So to get a meaningful
+    // result in this case we pick the minimum value obtainable from dataflow and range analysis.
+    result =
+      upperBound(statedSizeExpr())
+          .minimum(any(Expr statedSizeSrc |
```

Comment:
Should we be using the `min` aggregate here, rather than `int::minimum` and `any`?

Reply:
Good point. Fixed in 741eed93b2ddde49d69413cdac4d4511018f40d8.